src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.android.cs

```diff
@@ -22,18 +28,106 @@ public class MauiMediaElement : CoordinatorLayout
+		playerView.FullscreenButtonClick += OnFullscreenButtonClick;
+		this.playerView.SetBackgroundColor(Android.Graphics.Color.Black);
+
+		AddView(playerView);
+	}
+
+	/// <summary>
+	/// Allows the video to enter or exist a full screen mode
+	/// </summary>
+	/// <param name="sender"></param>
+	/// <param name="e"></param>
+	void OnFullscreenButtonClick(object? sender, StyledPlayerView.FullscreenButtonClickEventArgs e)
+	{
+		// Ensure there is a player view
+		if (this.playerView is null)
+		{
+			return;
+		}
+
+		// Ensure current activity and window are available
+		if (Platform.CurrentActivity is not Activity currentActivity
+			|| currentActivity.Window is not Android.Views.Window currentWindow
+			|| currentActivity.Resources is not Resources currentResources
+			|| currentResources.Configuration is null)
+		{
+			return;
+		}
+
+		var windowInsetsControllerCompat = WindowCompat.GetInsetsController(currentWindow, currentWindow.DecorView);
+
+		if (e.IsFullScreen)
+		{
+			// Force the landscape on the device
+			currentActivity.RequestedOrientation = Android.Content.PM.ScreenOrientation.Landscape;
+
+			// Hide the SystemBars and Status bar
+			if (OperatingSystem.IsAndroidVersionAtLeast(30))
+			{
+				currentWindow.SetDecorFitsSystemWindows(false);
+
+				var windowInsets = currentWindow.DecorView.RootWindowInsets;
+				if (windowInsets is not null)
+				{
+					isSystemBarVisible = windowInsets.IsVisible(WindowInsetsCompat.Type.NavigationBars()) || windowInsets.IsVisible(WindowInsetsCompat.Type.StatusBars());
+					
+					if (isSystemBarVisible)
+					{
+						windowInsetsControllerCompat.SystemBarsBehavior = WindowInsetsControllerCompat.BehaviorShowTransientBarsBySwipe;
+						currentWindow.InsetsController?.Hide(WindowInsets.Type.SystemBars());
+					}
+				}
+			}
+			else
+			{
+				defaultSystemUiVisibility = (int)currentWindow.DecorView.SystemUiVisibility;
+				int systemUiVisibility = defaultSystemUiVisibility | (int)SystemUiFlags.LayoutStable | (int)SystemUiFlags.LayoutHideNavigation | (int)SystemUiFlags.LayoutHideNavigation |
+					(int)SystemUiFlags.LayoutFullscreen | (int)SystemUiFlags.HideNavigation | (int)SystemUiFlags.Fullscreen | (int)SystemUiFlags.Immersive;
+				currentWindow.DecorView.SystemUiVisibility = (StatusBarVisibility)systemUiVisibility;
+			}
+
+			// Update the PlayerView
+			if (currentWindow.DecorView is FrameLayout layout)
+			{
+				RemoveView(playerView);
+				layout.AddView(playerView, new FrameLayout.LayoutParams(ViewGroup.LayoutParams.MatchParent, ViewGroup.LayoutParams.MatchParent));
+			}
+		}
+		else
 		{
-			LayoutParameters = new CoordinatorLayout.LayoutParams(LayoutParams.MatchParent, LayoutParams.MatchParent)
+			currentActivity.RequestedOrientation = Android.Content.PM.ScreenOrientation.Portrait;
+			
+			// Show again the SystemBars and Status bar
+			if (OperatingSystem.IsAndroidVersionAtLeast(30))
+			{
+				currentWindow.SetDecorFitsSystemWindows(true);
```

Comment:
Should we be caching this value before we go into `IsFullScreen == true` so that we only reset it back if it needs to be?

Reply:
Thanks @bijington for the suggestion. It is now being cached.